src/pages/admin.vue

```diff
@@ -63,26 +62,21 @@ function editUser(username: string) {
   };
   edittingUsername.value = username;
 }
+
+const {
+  isLoading,
+  errorMsg,
+  execute: modifyUser,
+} = useAsync(async () => {
+  if (!userForm.value.password) userForm.value.password = null;
+  await api.User.modify(edittingUsername.value, { ...userForm.value });
+  fetchUsers();
+  userForm.value = { ...initialUserForm };
+  edittingUsername.value = "";
+});
 async function submit() {
   if (!(await v$.value.$validate())) return;
-
-  isLoading.value = true;
-  try {
-    if (!userForm.value.password) userForm.value.password = null;
-    await api.User.modify(edittingUsername.value, { ...userForm.value });
-    execute();
-    userForm.value = { ...initialUserForm };
-    edittingUsername.value = "";
-  } catch (error) {
-    if (axios.isAxiosError(error) && error.response?.data?.message) {
-      errorMsg.value = error.response.data.message;
-    } else {
-      errorMsg.value = "Unknown error occurred :(";
-    }
-    throw error;
-  } finally {
-    isLoading.value = false;
-  }
+  modifyUser();
```

Comment:
Do we need to throw Error manually if there's errorMsg?

Reply:
error will be thrown in `useAsyncState`, we don't need to handle it

Reply:
Ok, I see that now.